tests/test_integrated_channels/test_sap_success_factors/test_transmitters/test_course_metadata.py

```diff
@@ -45,89 +48,90 @@ def setUp(self):
         )
-        self.send_course_import_mock = send_course_import_mock.start()
-        self.addCleanup(send_course_import_mock.stop)
+        self.create_course_content_mock = create_course_content_mock.start()
+        self.addCleanup(create_course_content_mock.stop)
 
-    def test_transmit_success(self):
+    @ddt.data(
+        (
+            (200, '{"success":"true"}'),
+            '200',
+            '',
+            {},
+            False
+        ),
+        (
+            RequestException('error occurred'),
+            '500',
+            'error occurred',
+            {},
+            False
+        ),
+        (
+            (200, '{"success":"true"}'),
+            '200',
+            '',
+            {
+                'test_course': {
+                    'in_catalog': True,
+                    'status': 'ACTIVE',
+                }
+            },
+            True
+        ),
+    )
+    @ddt.unpack
+    def test_transmit_success(
+            self,
+            expected_transmit_result,
+            expected_status,
+            expected_error_message,
+            audit_summary,
+            has_previous_transmission
+    ):
         """
-        The catalog data transmission audit that gets saved after the transmission completes contains success data.
+        The catalog data transmission audit that gets saved after the transmission completes contains appropriate data.
         """
-        self.send_course_import_mock.return_value = 200, '{"success":"true"}'
-        course_exporter_mock = mock.MagicMock(courses=self.payload)
-        course_exporter_mock.export.return_value = [(json.dumps(self.payload), 'POST')]
-        course_exporter_mock.resolve_removed_courses.return_value = {}
-        transmitter = course_metadata.SapSuccessFactorsCourseTransmitter(self.enterprise_config)
-        transmitter.transmit(course_exporter_mock)
-        self.send_course_import_mock.assert_called_with(json.dumps(self.payload))
-        course_exporter_mock.resolve_removed_courses.assert_called_with({})
-        course_exporter_mock.export.assert_called()
-        catalog_transmission_audit = CatalogTransmissionAudit.objects.filter(
-            enterprise_customer_uuid=self.enterprise_config.enterprise_customer.uuid,
-            channel=self.enterprise_config.provider_id,
-        ).latest('created')
-        assert catalog_transmission_audit.enterprise_customer_uuid == self.enterprise_config.enterprise_customer.uuid
-        assert catalog_transmission_audit.total_courses == len(self.payload)
-        assert catalog_transmission_audit.status == '200'
-        assert catalog_transmission_audit.error_message == ''
+        if has_previous_transmission:
+            audit_summary = {
+                'test_course': {
+                    'in_catalog': True,
+                    'status': 'ACTIVE',
+                }
+            }
```

Comment:
@UmanShahzad Do we need to define this again, or could we just reference `audit_summary` that gets passed in from `ddt`?

Reply:
Ah sorry, nice catch, this shouldn't be here :+1: